include/hipSYCL/glue/generic/hiplike/hiplike_kernel_launcher.hpp

```diff
@@ -30,24 +30,27 @@
 
 #include <cassert>
 
-#include "hipSYCL/sycl/backend/backend.hpp"
-#include "hipSYCL/sycl/range.hpp"
-#include "hipSYCL/sycl/id.hpp"
-#include "hipSYCL/sycl/item.hpp"
-#include "hipSYCL/sycl/nd_item.hpp"
-#include "hipSYCL/sycl/group.hpp"
+#include "hipSYCL/sycl/libkernel/backend.hpp"
+#include "hipSYCL/sycl/libkernel/range.hpp"
+#include "hipSYCL/sycl/libkernel/id.hpp"
+#include "hipSYCL/sycl/libkernel/item.hpp"
+#include "hipSYCL/sycl/libkernel/nd_item.hpp"
+#include "hipSYCL/sycl/libkernel/group.hpp"
+#include "hipSYCL/sycl/libkernel/detail/thread_hierarchy.hpp"
 
 #include "hipSYCL/runtime/device_id.hpp"
 #include "hipSYCL/runtime/kernel_launcher.hpp"
 
+#include "clang.hpp"
+
 
 namespace hipsycl {
 namespace glue {
 
 namespace hiplike_dispatch {
 
 template<int dimensions>
-__device__
+__host__ __device__
```

Comment:
any reason for not using `HIPSYCL_KERNEL_TARGET` here?

Reply:
It's not really necessary. `HIPSYCL_KERNEL_TARGET` is there to share code between backends that require specific attributes (looking at you, CUDA/HIP) and backends that don't. Since the kernel launchers are the heart of backend-specific kernel launching logic, we don't need to be more general here and can instead just use directly the attributes that we need.